src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Index.java

```diff
@@ -83,13 +84,14 @@ public ParsingFailedException(Path srcFile, ErrorCode code) {
 
     public TranslationUnit parseTU(String file, Consumer<Diagnostic> dh, int options, String... args)
     throws ParsingFailedException {
-        try (NativeScope scope = NativeScope.unboundedScope()) {
+        try (ResourceScope scope = ResourceScope.newConfinedScope()) {
```

Comment:
Watch out for indentation in this method - e.g. the `throws` clause seems odd on the new line. Also, there are missing spaces before/after the conditional operator `?`